rust/src/go/nmstate/nmstate.go

```diff
@@ -0,0 +1,199 @@
+// +build cgo
+package nmstate
+
+// #cgo CFLAGS: -g -Wall
+// #cgo LDFLAGS: -lnmstate
+// #include <nmstate.h>
+// #include <stdlib.h>
+import "C"
+import (
+	"fmt"
+	"io"
+	"time"
+)
+
+type Nmstate struct {
+	Timeout	    uint
+	LogsWritter io.Writer
+	Flags	    byte
```

Comment:
We don't hvae to expose this since they will be set with the `With` function.

Reply:
Done!